docs/tutorials/from-tabular-data-to-binned-data.ipynb

```diff
@@ -0,0 +1,679 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "5e7661da-24fc-434e-b0ec-75ce2b22af0f",
+   "metadata": {
+    "tags": []
+   },
+   "source": [
+    "# From tabular data to binned data"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "faa8d831-a5c1-44e4-b3be-602f23cd1dbd",
+   "metadata": {
+    "tags": []
+   },
+   "source": [
+    "## Overview"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "3eb1e88d-4725-4dea-8eb6-ace2514c4b06",
+   "metadata": {
+    "tags": []
+   },
+   "source": [
+    "Binned data in scipp is conceptually equivalent to an array of tables.\n",
+    "In other words, it represents an array of records (table rows) sorted into an (often multi-dimensional) array of \"bins\".\n",
+    "In this tutorial we begin by learning how to setup tabular data appropriate for histogramming and binning with scipp.\n",
+    "The main focus will then be binning the tabular data and basic usage of the resulting binned data.\n",
+    "\n",
+    "We will use a file of a simulated neutron-scattering experiment &mdash; at the powder diffractometer *DREAM* at the European Spallation Source.\n",
+    "The approach and techniques displayed here are however applicable for more generally and not specific to this scientific area."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "27a8267c-b5d1-4be3-9379-b4df15bb94dd",
+   "metadata": {
+    "tags": []
+   },
+   "source": [
+    "## Loading tabular data"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "a9ddb89f-4ad3-402f-93d0-1ef2cbb06428",
+   "metadata": {
+    "tags": []
+   },
+   "source": [
+    "We will use a file created by a McStas simulation for a diamond sample, with Geant4 simulating the detectors.\n",
+    "We can use [pandas.read_table](https://pandas.pydata.org/docs/reference/api/pandas.read_table.html) to load the table as a [pandas.Dataframe](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.html).\n",
+    "`pandas.read_table` is much faster than [numpy.loadtxt](https://numpy.org/doc/stable/reference/generated/numpy.loadtxt.html) and also slightly more convenient:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "b52a76b6-75ff-42b9-ad1f-843c7d4de5ba",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import pandas as pd\n",
+    "\n",
+    "filename = 'https://public.esss.dk/groups/scipp/scipp/1/data_dream_diamond.zip'\n",
+    "df = pd.read_table(filename)\n",
+    "df"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "4b53eb28-08e9-4085-a4be-f96db3da5ca5",
+   "metadata": {},
+   "source": [
+    "`scipp.compat.from_pandas` can convert the `pandas.Dataframe` to a `scipp.Dataset`.\n",
+    "The column names encode the physical units so we must extract them manually:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "64ef53f4-c6d2-497c-af60-b25e28beee46",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import scipp as sc\n",
```

Comment:
I think moving the import to the top is a bit cleaner.

Reply:
Where? Why?

Reply:
I think it makes more sense to put it at first use.